### ISSUE TITLE
Running train_with_point_labe.ipynb on cumstom data, the predicted cell mask is much larger than actual shape

### ISSUE THREAD

**Issue by lphilomena** [jiyuuchc/lacss#18]
Hi,

I recently tried running train_with_point_label.ipynb on custom data using the newest LACSS version, but the predicted cell masks appear significantly larger than the actual cell shapes. I tested both Lacss.get_small_model() and Lacss.get_large_model().

Could you clarify whether these models are trained from scratch or if they leverage pretrained weights from the LiveCell dataset?

For context, last year I ran the same notebook on an older LACSS version, finetuning the pretrained model cnsp4_bf on the same custom data, and the predicted masks aligned well with the actual cell shapes.

Do you have any insights into why this discrepancy might occur in the newer version?

Issue Example:

![Image](https://github.com/user-attachments/assets/c20f65bc-26dc-4557-bccd-d95a48c3a89e)

Thank you for your help!

**Comment by lphilomena** [comment]
Hi, I have tried this "config = ConfigDict()
config.max_training_instances = 3500
CKS.config.w = 0.01
model, params = load_from_pretrained('/root/project/lacss/pre_trained_models/lacss3-small-l')
# # Ensure the dictionary has the expected keys
trainer = Trainer(
    model = model,
    # model,
    optimizer = optax.adam(1e-4),
    losses = [], # losses is ignored by the CKS module
    strategy = CKS, # The CKS module implements the training logic
)" to fine-tune acss3-small-l on custome data and got training output 

" 10%|▉         | 2999/30000 [34:37<2:33:57,  2.92it/s] 
{'lpn_detection_loss': Array(0.0311706, dtype=float32), 'lpn_localization_loss': Array(0.13479914, dtype=float32), 'cks_loss': Array(0.1986481, dtype=float32)}
 20%|█▉        | 5999/30000 [52:50<2:18:05,  2.90it/s]  
{'lpn_detection_loss': Array(0.02664698, dtype=float32), 'lpn_localization_loss': Array(0.09302273, dtype=float32), 'cks_loss': Array(0.15778266, dtype=float32)}
 30%|██▉       | 8999/30000 [1:12:06<2:12:46,  2.64it/s] 
{'lpn_detection_loss': Array(0.02530294, dtype=float32), 'lpn_localization_loss': Array(0.07166941, dtype=float32), 'cks_loss': Array(0.167256, dtype=float32)}
 40%|███▉      | 11999/30000 [1:33:27<1:57:18,  2.56it/s] 
{'lpn_detection_loss': Array(0.0236186, dtype=float32), 'lpn_localization_loss': Array(0.05554964, dtype=float32), 'cks_loss': Array(0.174686, dtype=float32)}
 50%|████▉     | 14999/30000 [1:54:53<1:41:38,  2.46it/s]
{'lpn_detection_loss': Array(0.02193955, dtype=float32), 'lpn_localization_loss': Array(0.04376814, dtype=float32), 'cks_loss': Array(0.18379907, dtype=float32)}
 60%|█████▉    | 17999/30000 [2:16:26<1:09:02,  2.90it/s]
{'lpn_detection_loss': Array(0.02019757, dtype=float32), 'lpn_localization_loss': Array(0.03532806, dtype=float32), 'cks_loss': Array(0.19122076, dtype=float32)}
 70%|██████▉   | 20999/30000 [2:38:33<1:16:04,  1.97it/s]
{'lpn_detection_loss': Array(0.01811033, dtype=float32), 'lpn_localization_loss': Array(0.02837525, dtype=float32), 'cks_loss': Array(0.1986494, dtype=float32)}
 80%|███████▉  | 23999/30000 [3:00:43<50:29,  1.98it/s]  
{'lpn_detection_loss': Array(0.01576346, dtype=float32), 'lpn_localization_loss': Array(0.02311768, dtype=float32), 'cks_loss': Array(0.20370708, dtype=float32)}
 90%|████████▉ | 26999/30000 [3:22:56<25:11,  1.98it/s]  
{'lpn_detection_loss': Array(0.01373426, dtype=float32), 'lpn_localization_loss': Array(0.01931573, dtype=float32), 'cks_loss': Array(0.20958064, dtype=float32)}
100%|█████████▉| 29999/30000 [3:44:32<00:00,  2.09it/s]
{'lpn_detection_loss': Array(0.0119325, dtype=float32), 'lpn_localization_loss': Array(0.01692777, dtype=float32), 'cks_loss': Array(0.21421811, dtype=float32)}
100%|██████████| 30000/30000 [3:44:32<00:00,  2.23it/s]

The predicted cell mask is still larger than the actual cell shape, as seen in 

![Image](https://github.com/user-attachments/assets/c1a685ab-5776-466f-8963-4fb8a0bad6b2)

For context, last year I ran the same notebook on an older LACSS version, finetuning the pretrained model cnsp4_bf on the same custom data, and the predicted masks aligned well with the actual cell shapes (GT in the above image).

Do you have any insights into why this discrepancy might occur in the newer version?

Thank you for your help!

**Comment by jiyuuchc** [comment]
Thanks for reporting this regression bug.

It would be helpful if I could reproduce this bug on my computer. Is your training dataset an openly available one?

**Comment by jiyuuchc** [comment]
@[lphilomena](https://github.com/lphilomena)

It was pointed out to me that the a recent update altered the default hyperparameter of the CKS module. This may be related the issue you are encountering here. If you are still looking for a solution, the patch below will revert the change:

``` diff
--- a/lacss/train/cks.py
+++ b/lacss/train/cks.py
@@ -38,7 +38,7 @@ class LacssCollaborator(nn.Module):
     """

     conv_spec: Sequence[int] = (32, 32)
-    unet_spec: Sequence[int] = (16, 32, 64)
+    unet_spec: Sequence[int] = (32, 64)

     @nn.compact
     def __call__(self, image: ArrayLike, *args, **kwargs) -> DataDict:
```
